packages/network-console-shared/src/file_io/native/environment-container-adapter.ts

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+    IEnvironmentAdapter,
+    IEnvironmentContainerAdapter,
+    IEnvironmentFormat,
+} from '../interfaces';
+import {
+    INCNativeEnvironment,
+    INCNativeEnvironmentFile,
+} from './environment-format';
+import { EnvironmentAdapter } from './environment-adapter';
+import BidiMap from '../../util/bidi-map';
+
+type ValidityCheckResult = {
+    valid: true;
+    parsed: INCNativeEnvironmentFile;
+} | {
+    valid: false;
+    error: string;
+};
+
+export class EnvironmentContainerAdapter implements IEnvironmentContainerAdapter {
+    public canContainMultipleEnvironments = true;
+    private _dirty: boolean;
+    private _current: INCNativeEnvironmentFile;
+    private _keyToIndex: BidiMap<string, number>;
+    private _nextKey: number;
+
+    constructor(
+        public readonly format: IEnvironmentFormat,
+        public readonly id: string,
+        private fileContents: string,
+    ) {
+        const root = EnvironmentContainerAdapter.isValidFile(fileContents);
+        if (!root.valid) {
+            throw new RangeError('Invalid file or error parsing: ' + root.error);
+        }
+
+        this._current = root.parsed;
+        this._dirty = false;
+        this._keyToIndex = new BidiMap();
+        this._nextKey = 0;
+
+        for (let index = 0; index < this._current.environments.length; index++) {
+            const entryId = `${id}/${this._nextKey++}`;
+            this._keyToIndex.set(entryId, index);
+        }
+    }
+
+    static isValidFile(fileContents: string): ValidityCheckResult {

```

Comment:
`isValidFile` and `ValidityCheckResult` --> `parseAndCheck` and `ParseResult`, consistent with the Collection implementation.